src/sst/core/main.cc

```diff
@@ -426,6 +426,9 @@ main(int argc, char *argv[])
     double start_graph_gen = sst_get_cpu_time();
     graph = new ConfigGraph();
 
+    // Need to initialize TimeLord before we use UnitAlgebra
+    Simulation::getTimeLord()->init(cfg.timeBase);
+
```

Comment:
The `setProgramOption` function needs to propagate to TimeLord.

````cpp
Config::setTimebase(const std::string& arg){
  timebase = arg;
  Simulation::getTimeLord()->init(cfg.timeBase);
}

Making this change fixes the issue and all tests now pass.  This is, however, not a great way to fix things. There's a vague comment that we need to initialize TimeLord before using UnitAlgebra. It's not clear to me why these depend on each other.

Reply:
Looking at the code, the dependency between TimeLord and UnitAlgebra is gone.  I think when it was first written it stored things based on the timebase, but that hasn't been true for a while.  We should be able to safely reinitialize the TimeLord during the python execution.  Nothing is converted into TimeConverters until after the python is complete.

Reply:
For now, I just moved the initialization until after the python completes.  Probably the better way to give access to the timebase in the python is to look directly at the config object, which you can already do.  I may add a specific function to get the timebase later if it is needed to make it easier.